### PR TITLE
[new opmath 3] Make `qml.Hamiltonian` alias for `qml.LinearCombination` with new op math using `__getattr__`

### PR DESCRIPTION
**Pull request by mudit2812**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: doc/releases/changelog-dev.md
Comment (mudit2812) [inline doc/releases/changelog-dev.md:122]:
```suggestion
  [(#5393)](https://github.com/PennyLaneAI/pennylane/pull/5393)
```

File: pennylane/ops/qubit/hamiltonian.py
Comment (Qottmann) [inline pennylane/ops/qubit/hamiltonian.py:192]:
```suggestion
                "operator arithmetic before using qml.ops.Hamiltonian.",
```

Comment (mudit2812) [reply pennylane/ops/qubit/hamiltonian.py:192]: If the user disables new op math they can just use `qml.Hamiltonian`

Comment (mudit2812) [reply pennylane/ops/qubit/hamiltonian.py:192]: The warning only be raised if someone tries to create a `Hamiltonian` with new op math enabled. If we're telling people to disable new op math then we don't need to raise the warning as there's no concern about mixing old and new op math.

Comment (Qottmann) [reply pennylane/ops/qubit/hamiltonian.py:192]: Yeah but if new opmath is enabled, `qml.Hamiltonian` will point to `LinearCombination` 🙃

Comment (mudit2812) [reply pennylane/ops/qubit/hamiltonian.py:192]: Yeah, but in the warning I'm saying that to use the legacy Hamiltonian without the warning you have to `qml.operation.disable_new_opmath()` before using `qml.Hamiltonian`

Comment (astralcai) [reply pennylane/ops/qubit/hamiltonian.py:192]: I find this message somewhat confusing too because the reason the user is getting this message is precisely because they were using `qml.ops.Hamiltonian` instead of `qml.Hamiltonian`. If they were using `qml.Hamiltonian` with new op math enabled they would've gotten `LinearCombination`.

Comment (astralcai) [reply pennylane/ops/qubit/hamiltonian.py:192]: I guess the first sentence of the warning message should probably be "Using qml.ops.Hamiltonian with new operator arithmetic enabled is deprecated."

Comment (lillian542) [reply pennylane/ops/qubit/hamiltonian.py:192]: How about something like:

```
"Using the Hamiltonian class with new operator arithmetic is deprecated. 
Some functionality may not work as expected, and use of the Hamiltonian class with 
the new operator arithmetic will not be supported in future releases of PennyLane.

You can update your code to the new operator arithmetic by using qml.Hamiltonian 
instead of importing the Hamiltonian class directly. When the new operator arithmetic is enabled, 
`qml.Hamiltonian` will access the new corresponding implementation. 

Alternatively, to continue accessing the legacy functionality, you can use 
'qml.operation.disable_new_opmath()'."
```

Comment (astralcai) [reply pennylane/ops/qubit/hamiltonian.py:192]: > How about something like:

Could the user have used `qml.ops.Hamiltonian` because they didn't know that it would be different from `qml.Hamiltonian`, as opposed to intentionally looking for the old `Hamiltonian`? Should we explain this as well? It's not so common that when you import the same thing from different places you get completely different things.

File: tests/test_operation.py
Comment (Qottmann) [inline tests/test_operation.py:2759]: That is the leak

Comment (mudit2812) [reply tests/test_operation.py:2759]: Nice catch, I'll get rid of it. Apologies for the hassle

File: tests/test_operation.py
Comment (astralcai) [inline tests/test_operation.py:2778]: This fixture was removed from the updated test case. Adding it back would probably fix it.

Comment (mudit2812) [reply tests/test_operation.py:2778]: I think it got lost while resolving merge conflicts. Nice catch.

File: pennylane/ops/qubit/hamiltonian.py
Comment (Qottmann) [reply pennylane/ops/qubit/hamiltonian.py]: they would only get it with using `qml.ops.Hamiltonian` becauase when opmath is active `qml.Hamiltonian` will point to `LinearCombination` and this is not hit

Comment (mudit2812) [reply pennylane/ops/qubit/hamiltonian.py]: Not when new op math is enabled, as that would return a `LinearCombination`

File: pennylane/transforms/hamiltonian_expand.py
Comment (mudit2812) [inline pennylane/transforms/hamiltonian_expand.py:118]: Small change that was missed in the PR that fixed `hamiltonian_expand`.

File: doc/releases/changelog-dev.md
Comment (astralcai) [inline doc/releases/changelog-dev.md:128]: Could this potentially be a Breaking Change if someone creates a `qml.Hamiltonian` and expects it to be an instance of the old `Hamiltonian`?

File: doc/development/deprecations.rst
Comment (mudit2812) [inline doc/development/deprecations.rst:38]: Do we want to hide the fact that the new returned operator is `LinearCombination` from users?

Comment (mudit2812) [inline doc/development/deprecations.rst:38]: Do we want to hide the fact that the new returned operator is `LinearCombination` from users?

```suggestion
* Accessing ``qml.ops.Hamiltonian`` with new operator arithmetic is deprecated. Some functionality
```

Comment (astralcai) [reply doc/development/deprecations.rst:38]: I agree, the user probably doesn't have to know what a `LinearCombination`, or that it is part of new opmath.

File: doc/releases/changelog-dev.md
Comment (mudit2812) [reply doc/releases/changelog-dev.md:128]: Good point. I'll move it to the breaking changes section.

**Review by lillian542 — APPROVED** [review]
Looks great to me 🚀 🚀

File: doc/development/deprecations.rst
Comment (mudit2812) [reply doc/development/deprecations.rst:44]: I think it makes sense to say that `qml.Hamiltonian` will access the new implementation as defined by `qml.ops.LinearCombination`. However, this depends on decision we make about [the comment above this one](https://github.com/PennyLaneAI/pennylane/pull/5393#discussion_r1530532605). cc @trbromley

**Review by astralcai — APPROVED** [review]
🎸

File: doc/development/deprecations.rst
Comment (trbromley) [reply doc/development/deprecations.rst:38]: Let's keep mentioning `LinearCombination` here. One thing we might want to move away from in the docs and QML demos is using "Hamiltonian" out of context. That is, we will talk about Hamiltonians as observables determining energy in a system, but in other cases it might suffice to talk about "linear combinations".

**Review by trbromley — COMMENTED** [review]
Let's not block merging this PR with questions about the deprecations page. In the first month of Q2, we have a goal to ensure that our messaging for new op arithmetic is on point - which can include revisiting this deprecations page as well as considering having a quickstart page to explain what is happening.

**Review by Qottmann — APPROVED** [review]
Thanks @mudit2812 looks great!

Before merging, can you pull from `ham-tests` to see if all tests still pass (there have been some updates coming from the code suggestions in `LinearCombination` that have been pushed upstream)

**Pull request merged**
